amp/src/test/java/org/digijava/kernel/ampapi/endpoints/activity/FieldsEnumeratorTest.java

```diff
@@ -142,32 +150,36 @@ public void testComposition() {
         @Interchangeable(fieldTitle = "field_not_required_implicit")
         private String fieldNotRequiredImplicit;
 
-        @Interchangeable(fieldTitle = "field_not_required_explicit", required = ActivityEPConstants.REQUIRED_NONE)
+        @Interchangeable(fieldTitle = "field_not_required_explicit", required = NONE)
         private String fieldNotRequiredExplicit;
 
-        @Interchangeable(fieldTitle = "field_required_always", required = ActivityEPConstants.REQUIRED_ALWAYS)
+        @Interchangeable(fieldTitle = "field_required_always", required = ALWAYS)
         private String fieldRequiredAlways;
 
-        @Interchangeable(fieldTitle = "field_required_non_draft", required = ActivityEPConstants.REQUIRED_ND)
+        @Interchangeable(fieldTitle = "field_required_non_draft", required = SUBMIT)
         private String fieldRequiredNonDraft;
 
-        @Interchangeable(fieldTitle = "field_required_fm_visible", required = "fm visible")
+        @Interchangeable(fieldTitle = "field_required_fm_visible", requiredFmPath = "fm visible")
         private String fieldRequiredFmEntryVisible;
 
-        @Interchangeable(fieldTitle = "field_required_fm_hidden", required = "fm hidden")
+        @Interchangeable(fieldTitle = "field_required_fm_hidden", requiredFmPath = "fm hidden")
         private String fieldRequiredFmEntryHidden;
+    
+        @Interchangeable(fieldTitle = "disaster_response", fmPath = "fm visible",
+                requiredFmPath = "fm visible", required = ALWAYS)
+        private String fieldRequiredParentFmEntryVisible;
```

Comment:
Please add test for case when `requiredFmPath = "fm visible", required = DRAFT`.

Reply:
And `requiredFmPath = "fm hidden", required = DRAFT` and `requiredFmPath = "fm hidden", required = SUBMIT`.